packages/block-library/src/button/edit.js

```diff
@@ -81,19 +83,32 @@ function ButtonEdit( props ) {
 	const { textAlign, linkTarget, placeholder, rel, style, text, url, width } =
 		attributes;
 
-	function onToggleOpenInNewTab( value ) {
-		const newLinkTarget = value ? '_blank' : undefined;
+	function updateLinkAttributes( opensInNewWindow, nofollow ) {
+		let newLinkTarget;
+		let updatedRel = rel || '';
 
-		let updatedRel = rel;
-		if ( newLinkTarget && ! rel ) {
-			updatedRel = NEW_TAB_REL;
-		} else if ( ! newLinkTarget && rel === NEW_TAB_REL ) {
-			updatedRel = undefined;
+		if ( opensInNewWindow ) {
+			newLinkTarget = NEW_TAB_TARGET;
+			updatedRel = updatedRel?.includes( NEW_TAB_REL )
+				? updatedRel
+				: updatedRel + ` ${ NEW_TAB_REL }`;
+		} else {
+			const relRegex = new RegExp( `\\b${ NEW_TAB_REL }\\s*`, 'g' );
+			updatedRel = updatedRel?.replace( relRegex, '' ).trim();
+		}
+
+		if ( nofollow ) {
+			updatedRel = updatedRel?.includes( NOFOLLOW_REL )
+				? updatedRel
+				: updatedRel + ` ${ NOFOLLOW_REL }`;
+		} else {
+			const relRegex = new RegExp( `\\b${ NOFOLLOW_REL }\\s*`, 'g' );
+			updatedRel = updatedRel?.replace( relRegex, '' ).trim();
```

Comment:
Whist potentially "cleaner" I found this code much harder to read and understand than what we have in `inline.js`.

I wonder whether the simplest approach would be easier to follow in future?

Reply:
Unlike the inline link editor, the `rel` attr of Button link is editable, therefore when updating `rel` using LinkControl, we have to take the current `rel` attribute value into account (like skipping insert when existing, removing when the checkbox is false)

Reply:
I think it's fine then. Perhaps add a comment to explain that `rel` is editable as plaintext.

Reply:
Something like https://github.com/WordPress/gutenberg/pull/54110/files#diff-e335a5733e9910d51f3adb7c21b371b738fdc190e6b9b71b19ba2ad45703fea1R107